check-approved-requests.py

```diff
@@ -201,13 +201,16 @@ def check_requests(request_type, auth_file, worksheet_key):
                             **request_info)
  
             processed_rows.append(idx)
-            timestamp_spreadsheet(sheet, timestamp, processed_rows)
             if args.log:
                 log_request(args.log, timestamp, request_info['user_email'])
         else:
             # skip over unapproved or already-notified rows
             continue
 
+    # Google API returns an error if you send an empty request
+    if processed_rows:
```

Comment:
This script is checking every hour whether there are new approved requests, so frequently it doesn't find any.  In that case `processed_rows` is empty.

Without the if statement, it sends an empty request to the Google API and gets back an error.  Instead this lets the script realize it does not need to connect to Google this time, which is more efficient.

The "else" is "do nothing.".  This script runs as a cronjob so we don't print any output.  Eventually there will be logging and we might have something like "else log that there were no requests found."

Later if we implement logging there might be an else.